geotrek/api/v2/filters.py

```diff
@@ -269,48 +254,64 @@ def get_schema_fields(self, view):
         )
 
 
-class NearbyContentFilter(BaseFilterBackend):
+def _filter_near(base_model, target_model, target_pk, queryset):
+    """Filter the queryset of base_model objects by keeping only the objects near the target.
 
-    def intersect_queryset_with_object(self, qs, model, obj_pk, distance=None, field="geom"):
-        obj = model.objects.filter(pk=obj_pk).first()
-        if obj:
-            qs = intersecting(qs, obj, distance=distance, field=field)
-        else:
-            # Intersecting with a non-existing object results in empty data
-            qs = model.objects.none()
-        return qs
+    The function uses the model properties to achieve the filtering. For instance it would find and use the `target_trek.pois` property to filter
+    q POI queryset near a target trek.
 
-    def filter_queryset(self, request, qs, view):
-        distance = None
-        field = "geom"
+    Return an empty queryset if the target does not exist.
+    """
+
+    def pluralize(name):
+        return name + 's'
+
+    def get_prop(target_model, prop_name):
+        """Return the property object named 'prop_name' onto the target model or its parent classes
+        following the Method Resolution Order."""
+        for klass in (target_model,) + target_model.__mro__:
+            if prop_name in klass.__dict__:
+                return klass.__dict__[prop_name]
+        raise AttributeError(
```

Comment:
not covered by a test

Reply:
Good point. I have moved the function to `geotrek.api.v2.utils` and have added tests.